homeassistant/components/teltonika/sensor.py

```diff
@@ -0,0 +1,209 @@
+
+        if new_modems:
+            entities = [
+                TeltonikaSensorEntity(
+                    coordinator,
+                    data.device_info,
+                    description,
+                    modem_id,
+                    coordinator.data[modem_id],
+                )
+                for modem_id in new_modems
+                for description in SENSOR_DESCRIPTIONS
+            ]
+            async_add_entities(entities)
+            known_modems.update(new_modems)
+
+    # Add sensors for initial modems
+    _async_add_new_modems()
+
+    # Listen for new modems
+    entry.async_on_unload(coordinator.async_add_listener(_async_add_new_modems))
+
+
+class TeltonikaSensorEntity(CoordinatorEntity, SensorEntity):
+    """Teltonika sensor entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator,
+        device_info,
+        description: SensorEntityDescription,
+        modem_id: str,
+        modem: Any,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._modem_id = modem_id
+        self._attr_device_info = device_info
+
+        # Create unique ID using entry unique identifier, modem ID, and sensor type
+        entry_unique_id = (
+            coordinator.config_entry.unique_id or coordinator.config_entry.entry_id
+        )
+        self._attr_unique_id = f"{entry_unique_id}_{modem_id}_{description.key}"
+
+        # Use translation key for proper naming
+        modem_name = getattr(modem, "name", f"Modem {modem_id}")
+        self._modem_name = modem_name
+        self._attr_translation_key = description.translation_key
+        self._attr_translation_placeholders = {"modem_name": modem_name}
+        self._attr_suggested_object_id = slugify(
+            f"{modem_name} {description.translation_key or description.key}"
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return super().available and self._modem_id in self.coordinator.data
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        # Always set base attributes (all keys must always be present)
+        attrs = {
+            "modem_id": self._modem_id,
+            "modem_name": self._modem_name,
+            "connection_type": None,
+            "operator": None,
+        }
+
+        # Add sensor-specific attributes with None defaults
+        if self.entity_description.key in ("rssi", "rsrp", "rsrq", "sinr"):
+            attrs.update(
+                {
+                    "band": None,
+                    "state": None,
+                }
+            )
+
+        if self._modem_id not in self.coordinator.data:
+            self._attr_native_value = None
+            self._attr_extra_state_attributes = attrs
```

Comment:
what do we have exrta state attributes for? ideally we don't use any and create entities for them all, unless they don't belong in the state machine